vault/logical_system.go

```diff
@@ -809,6 +834,41 @@ type SystemBackend struct {
 	Backend *framework.Backend
 }
 
+// handleCORSRead returns the current CORS configuration
+func (b *SystemBackend) handleCORSRead(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	corsConf := b.Core.corsConfig
+	if corsConf == nil {
+		return nil, errCORSNotConfigured
+	}
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"enabled":         corsConf.Enabled,
+			"allowed_origins": strings.Join(corsConf.AllowedOrigins, ","),
+		},
+	}, nil
+}
+
+// handleCORSUpdate sets the list of origins that are allowed
+// to make cross-origin requests and sets the CORS enabled flag to true
+func (b *SystemBackend) handleCORSUpdate(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	origins := d.Get("allowed_origins").([]string)
+
+	err := b.Core.corsConfig.Enable(origins)
```

Comment:
Certainly not desired. I'm going to handle that, however, in the Enable func. That update func is more or less a wrapper for CORSConfig.Enable.